### DISCUSSION TITLE
What are the use-cases for this library?

### DISCUSSION THREAD

**Discussion by hazem3500** [react-figma/react-figma#460] [Q&A]
I am not sure when should someone use this library, 
if my goal was to create designs in Figma then why not use Figma directly? Or are these components meant to be used both in production (web/mobile) and Figma so we can have one source of truth for the design?

**Comment by cometkim** [comment]
I don't think the value of this lies in cross-platform code sharing. It's like React Native. The "Learn once, Write anywhere" strategy is incredibly flexible.

I can achieve stateful object manipulation in a declarative way I'm already familiar with without much confusion. The same was true when using other renderers like [phelia](https://github.com/maxchehab/phelia) or [ink](https://github.com/vadimmedes/ink). Considering the amount of time it usually takes to achieve the same results as using this, the results are quite impressive.

I used this library to visualize color tokens that are managed in the code in Figma. I will also use this library when making Figma plugins in the future.

**Reply by hazem3500** [reply]
I'm not sure if I caught your point correctly 😅

Do you mean you use this library more for creating manageable stateful designs in Figma, or are you using it to document the components you use in production in Figma?

**Reply by cometkim** [reply]
The design is still owned by the designer.

However, design tokens are a kind of data, and once they are stable, managing them in code is efficient. In this case, you may wanna build a Figma plugin to reflect library data into Figma back. React Figma is a very useful tool for this kind of work.

I didn't try the component as it is difficult to stabilize compared to the token. My guess is that even if you share 100% of the React GUI Primitives, it will be difficult to match the actual app code.

In summary, to automate any kind of task in the Figma canvas, you can open a console and use the Figma API directly, or you can reuse the language you already know.
